custom_components/unifiprotect/binary_sensor.py

```diff
@@ -441,3 +441,70 @@ def _async_fire_motion_event(self) -> None:
     def extra_state_attributes(self) -> dict[str, Any]:
         """Return the extra state attributes."""
         return {**super().extra_state_attributes, **self._extra_state_attributes}
+
+
+class UnifiProtectAccessTokenBinarySensor(UnifiProtectBinarySensor, AccessTokenMixin):
+    def __init__(
+        self,
+        protect: ProtectApiClient,
+        protect_data: UnifiProtectData,
+        device: ProtectDeviceModel,
+        description: UnifiProtectBinaryEntityDescription,
+        index: int | None = None,
+    ) -> None:
+        assert isinstance(device, Camera)
+        self.device: Camera = device
+        super().__init__(protect, protect_data, device, description, index)
+
+    @callback
+    def _async_get_extra_attrs(self) -> dict[str, Any]:
+        attrs = super()._async_get_extra_attrs()
+
+        # Camera motion sensors with object detection
+        event: Event | None = None
+        score = 0
+        if (
+            self.device.is_smart_detected
+            and self.device.last_smart_detect_event is not None
+            and len(self.device.last_smart_detect_event.smart_detect_types) > 0
+        ):
+            score = self.device.last_smart_detect_event.score
+            event = self.device.last_smart_detect_event
+            detected_object: str | None = (
+                self.device.last_smart_detect_event.smart_detect_types[0].value
+            )
+            _LOGGER.debug(
+                "OBJECTS: %s on %s",
+                self.device.last_smart_detect_event.smart_detect_types,
+                self._attr_name,
+            )
+        else:
+            if (
+                self.device.is_motion_detected
+                and self.device.last_motion_event is not None
+            ):
+                score = self.device.last_motion_event.score
+            detected_object = None
+            event = self.device.last_motion_event
+
+        thumb_url: str | None = None
+        if event is not None:
+            assert self.device_info is not None
+            # thumbnail_id is never updated via WS, but it is always e-{event.id}
+            thumb_url = (
+                ThumbnailProxyView.url.format(event_id=f"e-{event.id}")
+                + f"?entity_id={self.entity_id}"
+            )
+            _LOGGER.debug("Event thumbnail URL for %s: %s", self.entity_id, thumb_url)
+            thumb_url += f"&token={self.access_tokens[-1]}"
```

Comment:
It would be cleaner to use `from urllib.parse import urlencode`

` urlencode({'entity_id':'x','token':'xxx'})`